packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts

```diff
@@ -33,9 +33,25 @@ export class R11sDocumentDeltaConnection extends DocumentDeltaConnection
                     tenantId,
                 },
                 reconnection: false,
-                transports: ["websocket"],
+                // Enable long-polling as a downgrade option when WebSockets are not available
+                transports: ["websocket", "polling"],
```

Comment:
Curious, does it matter if polling is added here, given that you change transports below (on error)? Or both places need to have it in the list?
Maybe other way to ask - if (for example) a host wants to control this behavior, where we would put IF condition to disable/enable long polling? Should we do it from the start (and bake into generic layer) instead of having it here, and each driver just control policy? Would be easier down the road for ODSP, and would allow not to replicate bugs across layers due to duplicated code.

Reply:
Putting it here allows for polling to be used in scenarios where websockets are not available in the given runtime (e.g. IE 9). I think currently IE 9 is the only browser that doesn't support websockets, and engine.io-client has built-in isomorphic support so it also works in Node.js via `ws` package. Realistically, adding this line here adds very niche support, but I figured it's not really hurting anything to add it. 

As for where to put an IF condition, I was under the impression that we didn't want to bake socket.io specifics into the generic layer in case a driver/service wanted to use a different WebSockets implementation, but that might be an outdated understanding. In the case that it's here in the per-driver layer, I believe the IF condition would have to go both around this config line as well as around the following `.on("connect_error")` handler, since that is really where the majority of polling usecases are covered.

Reply:
Do we care about IE9? I think we should not, unless there is some strong case here (I'm not aware of).

I'm not sure what you mean by different WebSocket implementation. If driver choses to use this class, then it agrees to use socket.io and appropriate transport protocols required / supported by this layer. It feels standard issue of code reuse - I do not see much value in duplicated same logic across drivers. If we expect same logic to be in other drivers that want long polling, then I'd rather move it to shared layer, and expose policy to enable / disable it.

I'd go even further - you want it for single driver to have ability to easily control it, and thus allow end experiences to evaluate / enable it slowly, observing impact (COGS, perf, etc.) and potentially cut it off through dynamic feature gates if things go wrong.

I can also see (maybe it's in some distant future) requests in the form of - could you please cache that given tenant does not support websockets, and thus move to long polling faster? I'd prefer to give tools to other layers (hosts) to be able to implement something like that, and simply expose enough policy controls for them to be able to do so, vs. bake it into drivers.

Reply:
> Do we care about IE9? I think we should not, unless there is some strong case here (I'm not aware of).

Nope, just now found this [FluidFramework browser support doc](https://github.com/microsoft/FluidFramework/blob/main/docs/_includes/browsers.md). I think I can safely remove this additional polling config, but will verify that everything still works.

> Other stuff

Totally agree. Thank you for explaining that driver-base's DocumentDeltaConnection requires socket.io (looking at the types now this should have been obvious). I will work on an update to this PR that moves this handling into the shared space.